### PR TITLE
fix(types): fix JSONValue type

### LINKED ISSUE #104: [TS][Types] Type error when returning deep nested data in defineEventHandler
https://stackblitz.com/edit/github-7pwass?file=server/api/test.ts

### PR DESCRIPTION
**Pull request by antfu**
https://www.typescriptlang.org/play?#code/PTAEDsFMHcCgEtwBdICcBmBDAxpUApAZQHkA5YgIwCtJslQBvWUUAbQA8AuUAZyVUQBzALrciZAGqYANgFdIAblgBfBMjRZcBEqQCCqVJgCeoSOxTgAJj1D7DRgDzjSUuZAB8jUKthmADgD2qPRIRn54zq7yoAC8zKAAPrz8QvFJ4LIAthRoaaAUAQHSkJjgec6UNHTlOnbGsLAgoADCsgaQyLCh4aAA+pEy8g4AKjF8AuCCCRnZaAkFRSXgnjGgw4lrrMIbAEq0QZYO40IANGvuvuyBwaDdETpRkMTSlrF9A24O-Q+DHg2w2AC4D4oEwYh+bjeTBYYNAAHJjpM4Sd4hRuKwGKB0IVuAiUkjvMIVA0mgBRAxBAFAkFo7SSX7PV6raGg3GIwTI1HozHYgJs-EcwnEoA

The current type does not handle nested object/array correctly

Close #104

Edit: oh it conflicts with #105, feel free to pick one you prefer :)

### CODE REVIEW

File: src/event.ts [hunk 1]
```diff
@@ -12,8 +12,14 @@ export interface H3Event {
 
 export type CompatibilityEvent = H3Event | IncomingMessage
 
-type _JSONValue<T=string|number|boolean> = T | T[] | Record<string, T>
-export type JSONValue = _JSONValue<_JSONValue>
+interface JSONObject { [x: string]: JSONValue }
+interface JSONArray extends Array<JSONValue> { }
+export type JSONValue =
+  | string
```
Comment (pi0) [inline src/event.ts:18]: Style: would you please make it one line too?

Comment (antfu) [reply src/event.ts]: Done

**Comment by replygirl** [comment]
closed mine thx :)

**Comment by pi0** [comment]
Thanks @antfu and @replygirl!

**Comment by Rigo-m** [comment]
@pi0 can we bump version so I don't have this nasty TS error showing up in my nuxt 3 projects? :P thanks, great job!

**Pull request merged**